lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1260,18 +1262,34 @@ def create_ReAuthRequest(self, imsi, policy_id, flow_list, qos):
         qos = QoSInformation(qci=qos["qci"])
 
         # Get sessionid
-        req = GetDirectoryFieldRequest(id=imsi, field_key="session_id")
-        try:
-            res = self._directorydstub.GetDirectoryField(
-                req, DEFAULT_GRPC_TIMEOUT
-            )
-        except grpc.RpcError as err:
-            logging.error(
-                "GetDirectoryFieldRequest error for id: %s! [%s] %s",
-                imsi,
-                err.code(),
-                err.details(),
-            )
+        #TODO: remove retries
+        i = 0
+        MAX = 3
+        while i < MAX:
+            req = GetDirectoryFieldRequest(id=imsi, field_key="session_id")
+            try:
+                res = self._directorydstub.GetDirectoryField(
+                    req, DEFAULT_GRPC_TIMEOUT
+                )
+            except grpc.RpcError as err:
+                print(
+                    "error: GetDirectoryFieldRequest error for id: %s! [%s] %s",
+                    imsi,
+                    err.code(),
+                    err.details(),
+                )
+            if req != None:
+                i = MAX
+            else:
+                i+=1
+                print("warning: directoryd failed to return sessionId for %s. Retrying", imsi)
+                time.sleep(3)
+
+        if req == None:
+            print("error: Couldnt find sessionid. Directoryd content:")
+            ret = self._directorydstub.GetAllDirectoryRecords(Void(), DEFAULT_GRPC_TIMEOUT)
+            for record in ret.records:
+                print("%s", str(record))
```

Comment:
why do we want to clutter the test output like this?

Reply:
This is a temporal function. Unfortunately the issue only happens on CI.

Once it is fixed i will remove it